deps/build.jl

```diff
@@ -45,56 +45,61 @@ end
 
 ## discovery routines
 
-"Return path of CUDA toolkit and the runtime library."
+# find CUDA toolkit
 function find_cuda()
-    cudapath_envs = ["CUDA_PATH", "CUDA_HOME", "CUDA_ROOT"] 
-    cudapath = Nullable{String}()
-    if any(x -> haskey(ENV, x), cudapath_envs)
-        cudapath_envs = cudapath_envs[map(x -> haskey(ENV, x), cudapath_envs)]
-        if !all(i -> i == cudapath_envs[1], cudapath_envs)
-            warn("Multiple, non-equivalent CUDA path variables found in environment.")
-            cudapath = Nullable(ENV[first(cudapath_envs)])
-            warn("Arbitrarily selecting CUDA at $(get(cudapath)). To ensure a consistent path, ensure only one CUDA path environment variable is set.")
-        else
-            cudapath = Nullable(ENV[first(cudapath_envs)])
+    cuda_envvars = ["CUDA_PATH", "CUDA_HOME", "CUDA_ROOT"]
+    cuda_envvars_set = filter(var -> haskey(ENV, var), cuda_envvars)
+    if length(cuda_envvars_set) > 0
+        cuda_paths = unique(map(var->ENV[var], cuda_envvars_set))
+        if length(unique(cuda_paths)) > 1
+            warn("Multiple, non-equivalent CUDA path variables found in environment. ",
+                 "Arbitrarily selecting CUDA at $(first(cuda_paths)). ",
+                 "To ensure a consistent path, ensure only one CUDA path environment variable is set.")
```

Comment:
Maybe mention which cuda environment vars are set? It is won't be clear for people what "one CUDA path" environment is.